### PR TITLE
add feature to get clubs from chain

### PR DESCRIPTION
**Pull request by QingyangKong**
Add method to get club info from chain.
Complete feature signup with mock data.

### CODE REVIEW

File: src/containers/Register/index.js [hunk 1]
```diff
@@ -18,19 +18,22 @@ class Register extends React.Component {
...
       handleRegister,
       isInfoCompleted,
+      checkIfRegistered,
+      accountSignUp,
     } = this.props.registerStore
 
     return (
       <div className='register__container'>
         <Header titleName='用户设置' backRoute='/user'/>
+        {/* <Button onClick={checkIfRegistered}>check if user signed up</Button> */}
```
Comment (wangwenyue) [inline src/containers/Register/index.js]: Better be deleted.

File: src/containers/Register/index.js [hunk 2]
```diff
@@ -18,19 +18,22 @@ class Register extends React.Component {
...
         <div className='register__container--content'>
           <RegisterAddress registerAddress={registerAddress}/>
           <RegisterName onChange={onRegisterAddressChange}/>
           <RegisterAvatar files={files} onChange={onRegisterAvatarChange}/>
           <ClubLogo/>
           <Button
-            disabled={!isInfoCompleted}
-            className={`register__button--finish ${!isInfoCompleted ? 'inactive' : ''}`} onClick={handleRegister}>
+            disabled={false}
```
Comment (wangwenyue) [inline src/containers/Register/index.js]: Use `!isInfoCompleted` instead of `false`

File: src/stores/registerStore/index.js [hunk 3]
```diff
@@ -28,6 +32,55 @@ class RegisterStore {
...
+    Promise.all([currentAddr, currentBlockNumber]).then(([currentAddress, blockNumber]) => {
+      const tx = {
+        ...transaction,
+        from: currentAddress,
+        validUntilBlock: blockNumber + 88,
+      }
+      const userContract = new appchain.base.Contract(playerAbi, config.userContract)
+      return userContract.methods.signIn('sampleName', 'sampleUrl').send(tx)
+    }).then(result => {
+      log('waiting for transactino result')
```
Comment (wangwenyue) [inline src/stores/registerStore/index.js]: typo `transaction`

File: src/containers/User/ClubListItem/index.js [hunk 4]
```diff
@@ -1,35 +1,65 @@
...
+@inject('clubListStore') @observer
 class ClubListItem extends React.Component {
 
+  constructor(props) {
+    super(props)
+    this.state = this.props.clubListStore
+  }
+
+  componentDidMount() {
+    this.state.getUserClubs(10)
```
Comment (Keith-CY) [inline src/containers/User/ClubListItem/index.js]: do not put function in state

File: src/containers/User/ClubListItem/index.js [hunk 5]
```diff
@@ -1,35 +1,65 @@
...
+      clubUsers,
+      clubUserAvatars,
+    } = this.state
+
+    let clubData = []
+    if (clubIdList.length === clubNameList.length) {
+      for (let i = 0; i < clubIdList.length; i++) {
+        clubData.push({
+          clubName: clubNameList[i],
+          avatar: clubUserAvatars.slice().map(proxy => proxy.slice()),
```
Comment (Keith-CY) [inline src/containers/User/ClubListItem/index.js:36]: what does `slice()` do?

File: src/containers/User/ClubListItem/index.js [hunk 6]
```diff
@@ -1,35 +1,65 @@
...
+          clubID: clubIdList[i],
+        })
+      }
+    }
 
-    const clubList = clubDataList.map((data, index) => {
+    const clubList = clubData.length === 0 ? clubData : clubData.map((data, index) => {
```
Comment (Keith-CY) [inline src/containers/User/ClubListItem/index.js]: unnecessary `clubData.length === 0 ? clubData : ...`

File: src/containers/User/ClubListItem/index.js [hunk 7]
```diff
@@ -1,35 +1,65 @@
...
-                  if (index <= maxAvatars) {
-                    return <li key={index}><img src={avatar} alt=""/></li>
-                  }
-                })}
-              </ul>
+              {
+                data.avatar ? 
+                  <ul>
+                    {
+                      data.avatar.map((avatar, index) => {
```
Comment (Keith-CY) [inline src/containers/User/ClubListItem/index.js]: `data.avatar.slice(0, maxAvatars + 1).map`

File: src/stores/clubListStore/index.js [hunk 8]
```diff
@@ -1,26 +1,77 @@
...
 
+  @action async getUserClubs(i) {
+    const userContract = new appchain.base.Contract(playerAbi, config.userContract)
+    const clubContract = new appchain.base.Contract(clubAbi, config.clubContract)
+
+    const sender = await appchain.base.getDefaultAccount()
+
+    const size = await userContract.methods.getUserClubsSize(sender).call()
+    
+    let max = size != 0? size : size
```
Comment (Keith-CY) [inline src/stores/clubListStore/index.js]: meaningless...

**Review by Keith-CY — CHANGES_REQUESTED** [review]

File: src/containers/User/ClubListItem/index.js [hunk 9]
```diff
@@ -1,35 +1,65 @@
 import React from 'react'
 import { inject, observer } from 'mobx-react'
+import { observe } from 'mobx'
```
Comment (wangwenyue) [inline src/containers/User/ClubListItem/index.js]: Not used import item.

File: src/containers/User/ClubListItem/index.js (hunk 5 above)
Comment (QingyangKong) [reply src/containers/User/ClubListItem/index.js:36]: `slice()` is to convert Proxy to values, every observable in store is saved as proxy.

**Review by Keith-CY — APPROVED** [review]

**Pull request merged**